tsconfig.base.json

```diff
@@ -0,0 +1,20 @@
+{
+  "compilerOptions": {
+    "allowJs": true,
+    "allowSyntheticDefaultImports": true,
+    "esModuleInterop": true,
+    "forceConsistentCasingInFileNames": true,
+    "importHelpers": true,
+    "inlineSources": true,
+    "jsx": "react",
+    "lib": ["es7", "es6", "dom", "dom.iterable"],
+    "module": "esnext",
+    "moduleResolution": "node",
+    "noImplicitReturns": true,
+    "noUnusedLocals": true,
+    "pretty": true,
+    "sourceMap": true,
+    "strict": true,
+    "target": "es5"
```

Comment:
Should we not set the target to `es6` or `esnext`?

Reply:
`"target": "es5"` will generate es5 code (so no classes, no template strings, etc).
`"module": "esnext"` will generate es6 imports/exports.
That way we get tree-shaking from bundlers + ie11 compat.

We could just not transpile at all besides ts types/enums, and let users transpile down to the browsers they target, wdyt?

Reply:
That is what I was thinking as well and let the users take care of transpiling. Then we probably not need tslib also. 

Just to be on the safer side, lets just keep ES5 for now and we can revisit.

Reply:
Sounds good.